dev/src/reference.ts

```diff
@@ -216,8 +223,12 @@ export class DocumentReference implements Serializable {
    *   console.log(`Found ${results.size} matches in parent collection`);
    * }):
    */
-  get parent(): CollectionReference {
-    return new CollectionReference(this._firestore, this._path.parent()!);
+  get parent(): CollectionReference<T> {
+    return new CollectionReference(
```

Comment:
I don't think you have to specify it in this case because `this._converter` is typed to `T` already, so it's automatically inferred.

Or is this a more style / code readability issue?

Reply:
Since we are explicitly specifying the return type, I would prefer if we also specify the type of the generic. To me this looks like it is using the default value (T=DocumentData), which is not always correct.